Leetcode/aoc_2015_d5.py

```diff
@@ -0,0 +1,105 @@
+import unittest
+
+'''
+ ----- Part 1 -----
+Given an input of strings separated by newlines, figure out how many of the strings are nice.
+A string is defined as nice when it has at least 3 vowels, a character that appears twice in a row, i.e.
+aa, dd, bb, etc.
+And does not contain the strings ab, cd, pq, or xy.
+Since there are only 4 strings that we need to make sure are not in the string, we can put that into
+a list and use a loop that checks the input string if it finds that particular sequence, if it does, move onto
+the next string.
+Can have a count variable that resets every new string. Count variable will check how many vowels are in the string.
+Put the vowels in a list, check each string character by character, if the character is in the vowel list, increment
+the count.
+We can also have a boolean variable that defaults to false every new string. While it's parsing the string char by char,
+if it finds a double letter, change to true.
+If there isn't an ab, cd, pq, or xy, vowel count is greater than 3, and double_char is true, add one to nice strings
+variable.
+'''
+
+
+with open("aoc_2015_d5_input.txt") as file:
+	content = [line.rstrip("\n") for line in file]
+
+
+def nice_string(str_list: list) -> int:
+	nice_strings = 0
+	for s in str_list:
+		double_char = False
+		vowel_count = 0
+		vowels = ['a', 'e', 'i', 'o', 'u']
+		# Decided to do this instead of a list that contains the banned strings because
+		# it'd require a nested loop and a variable. This way I can just continue the current loop.
+		if "ab" in s or "cd" in s or "pq" in s or "xy" in s:
+			continue
+		for i in range(len(s)):
+			if i < len(s) - 1 and s[i + 1] == s[i]:
+				double_char = True
+			if s[i] in vowels:
+				vowel_count += 1
```

Comment:
Would it really make readability that much better?
It seems fine to me, could maybe use some whitespace.
I think if I saw two loops that were going through the same thing each with only one conditional, I'd be questioning why they were separate.

Reply:
Well the proper deconstruction of it would be to separate it into two functions - one `doublechar` function, and one `vowel_count` function.

But the bigger answer is that I probably won't do a manual loop for the vowel count, I'd use a Counter.

The reason why I want to see doublechar in its own loop is because it actually doesn't loop to the end of the loop because of the `i < len(s) - 1` - semantically, they are two different loops, and thus should be written separately.

Reply:
Split into different loops.